includes/wcs-order-functions.php

```diff
@@ -386,6 +387,18 @@ function wcs_get_orders_with_meta_query( $args ) {
 		add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $handle_meta, 10, 2 );
 	}
 
+	/**
+	 * Map the 'any' status to wcs_get_subscription_statuses() in HPOS environments.
+	 *
+	 * In HPOS environments, the 'any' status now maps to wc_get_order_statuses() statuses. Whereas, in
+	 * WP Post architecture 'any' meant any status except for ‘inherit’, ‘trash’ and ‘auto-draft’.
+	 *
+	 * If we're querying for subscriptions, we need to map 'any' to be all valid subscription statuses otherwise it would just search for order statuses.
+	 */
+	if ( [ 'any' ] === $args['status'] && 'shop_subscription' === $args['type'] && wcs_is_custom_order_tables_usage_enabled() ) {
```

Comment:
The comment here is pretty self explanatory but just to explain some of the background. I discovered when I tried to search for subscriptions using this method on HPOS environments that it wasn't able to find them. In most cases we use to send the status arg as `array( 'any' )` but the HPOS query class doesn't support that format whereas the `WP_Query` one did. 

Unfortunately there isn't a clear replacement for the `WP_Query` `any` status arg. [`OrdersTableQuery::sanitize_status`](https://github.com/woocommerce/woocommerce/blob/7.1.0/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php#L455-L472) turns 'any' into the status returned from `wc_get_order_statuses()`. 
There is an `all` option, however `WP_Query` `any`  excludes `‘inherit’`, `‘trash’` and `‘auto-draft’`. ([ref](https://developer.wordpress.org/reference/classes/wp_query/#status-parameters)) by default and we wouldn't want to return trashed subscriptions.